geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/KryoFeatureSerializer.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.feature.kryo
+
+import java.io.{InputStream, OutputStream}
+
+import com.esotericsoftware.kryo.Kryo
+import com.esotericsoftware.kryo.io.{Input, Output}
+import org.locationtech.geomesa.feature.AvroSimpleFeature
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+/**
+ * Class for serializing and deserializing simple features. Not thread safe.
+ *
+ * @param sft
+ * @param decodeAs
+ */
+case class KryoFeatureSerializer(sft: SimpleFeatureType, decodeAs: SimpleFeatureType) {
+
+  private val serializer = if (sft.eq(decodeAs)) {
+    new SimpleFeatureSerializer(sft)
+  } else {
+    new TransformingSimpleFeatureSerializer(sft, decodeAs)
+  }
+
+  private val kryo = new Kryo()
+
+  kryo.setReferences(false)
+  kryo.register(classOf[KryoSimpleFeature], serializer,  kryo.getNextRegistrationId)
```

Comment:
I haven't tried spark... I will see if I can test it out. I believe that since this is a separate kryo instance it won't interfere with anything else.